### PR TITLE
fix: resolve rebuild-day exercise names

### PR DESCRIPTION
**Pull request by JeffOtano**
## Summary

- Fixes the weekly `rebuild_day` path so exact exercise names from `search_exercises` can repair missing, stale, or fabricated Tonal movement IDs.
- Prevents the coach from getting stuck when rebuilding a single day inside a week plan after movement ID validation failures.
- Keeps persisted workout blocks canonical by stripping temporary exercise names before saving the rebuilt draft.

## Changes

- Updated `rebuild_day` tool guidance/schema to accept exact exercise names and optional movement IDs.
- Added name-based resolution in `convex/coach/rebuildDay.ts` using the existing Tonal movement resolver before creating the replacement draft workout.
- Added a production-path regression test through `internal.coach.rebuildDay.rebuildDay` for fabricated ID repair.

## Checklist

- [x] `npx tsc --noEmit` passes
- [x] `npm run lint` passes
- [x] `npm test` passes (existing + new tests)
- [x] New logic has tests (happy path + at least one error/edge case)
- [x] No new or materially expanded file exceeds the 300-line soft cap
- [x] No file exceeds the 400-line hard cap (enforced by CI)
- [x] Functions stay near the 60-line convention
- [x] No new `any` types (enforced by ESLint); `as` casts only at deserialization boundaries
- [ ] Tested in browser (if UI changes)
- [x] Commits follow conventional format (`type: description`)
- [x] No unused exports or dead code introduced

## Test Plan

- `npx tsc --noEmit`
- `npx vitest run convex/coach/rebuildDay.test.ts`
- `npm test`
- `npm run lint`
- `npx prettier --check convex/ai/rebuildDayTool.ts convex/coach/rebuildDay.ts convex/coach/rebuildDay.test.ts`

No browser verification was needed because this is a Convex backend/tool-schema fix with no UI surface changes.

## Summary by CodeRabbit

* **Improvements**
  * Updated workout rebuilds to validate the incoming block/exercise structure before processing.
  * Improved exercise matching by resolving against the full movement catalog using exact exercise names (with clearer guidance on providing `name`).
  * When a movement is resolved, automatically normalizes reps vs duration.
  * Enhanced feedback for unresolved or ambiguous exercise matches.
* **Bug Fixes**
  * Added handling to fix fabricated movement identifiers via name-based resolution, and return validation errors instead of throwing.
* **Tests**
  * Expanded rebuild coverage with new fixtures and additional validation/matching scenarios.

### CODE REVIEW

File: convex/coach/rebuildDay.ts [hunk 1]
```diff
@@ -13,18 +13,67 @@ import { v } from "convex/values";
...
+        spotter: v.optional(v.boolean()),
+        eccentric: v.optional(v.boolean()),
+        chains: v.optional(v.boolean()),
+        burnout: v.optional(v.boolean()),
+        dropSet: v.optional(v.boolean()),
+        warmUp: v.optional(v.boolean()),
+      }),
+    ),
+  }),
+);
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/coach/rebuildDay.ts:53]: Good catch. I added `validateRebuildDayBlocks` in `convex/coach/rebuildDay.ts` so direct action calls now reject empty block/exercise arrays and invalid `sets`/`reps`/`duration` values before draft creation. I also covered those cases in `convex/coach/rebuildDay.test.ts` and verified with `npx tsc --noEmit`, `npx vitest run convex/coach/rebuildDay.test.ts`, `npm run lint`, and `npm test`. Resolving this thread.

**Comment by JeffOtano** [comment]
Addressed the CodeRabbit feedback in `e30d380f`:

- Added direct-action block validation in `convex/coach/rebuildDay.ts` for empty block/exercise arrays and invalid `sets`/`reps`/`duration` values.
- Added regression coverage in `convex/coach/rebuildDay.test.ts` for unresolved exercise-name guidance and malformed direct action inputs.
- Reworked the test fixtures to keep `convex/coach/rebuildDay.test.ts` below the 300-line soft cap.

Verified locally with `npx tsc --noEmit`, `npx vitest run convex/coach/rebuildDay.test.ts`, `npm run lint`, and `npm test`.

**Comment by JeffOtano** [comment]
Addressed the remaining CodeRabbit test-hygiene nitpicks in `5c94c891`:

- Replaced `Date.now()` fixture timestamps in `convex/coach/rebuildDay.test.ts` with a fixed timestamp.
- Replaced avoidable `as const` fixture casts with typed fixture helpers.
- Swapped error-result type assertions for `result.ok` control-flow narrowing.

Verified locally with `npx tsc --noEmit`, `npx vitest run convex/coach/rebuildDay.test.ts`, `npm run lint`, `npx prettier --check convex/coach/rebuildDay.test.ts convex/coach/rebuildDay.ts convex/ai/rebuildDayTool.ts`, and `npm test`.

@codex review
